frontend/src/components/InlineSetting/InlineSetting.tsx

```diff
@@ -18,37 +18,47 @@ type Props = {
   value?: string | number
   label: string
   icon?: string
+  displayValue?: string | number
+  iconTooltip?: string
   filter?: RegExp
   disabled?: boolean
   resetValue?: string | number
   onSave: (value: string | number) => void
 }
 
 export const InlineSetting: React.FC<Props> = ({
-  value,
+  value = '',
   label,
   icon,
+  iconTooltip,
+  displayValue,
   disabled,
   resetValue,
   onSave,
   filter,
   children,
 }) => {
   const [edit, setEdit] = useState<boolean>(false)
-  const [editValue, setEditValue] = useState()
+  const [editValue, setEditValue] = useState<string | number>('')
 
   const css = useStyles()
   const showEdit = () => {
     setEditValue(value)
     setEdit(true)
   }
 
+  const LeftIcon = (
+    <Tooltip open={iconTooltip ? undefined : false} title={iconTooltip || ''}>
```

Comment:
Maybe simpler, I've I'm understanding correctly?

```suggestion
    <Tooltip open={iconTooltip} title={iconTooltip}>
```

Reply:
Yes agree these are weird, but they are actually required for the feature to work - undefined and ' '  and false are all different behaviors.

Reply:
the `open` one is saying if there is a tooltip do default behavior of show/hide on hover, but do not open if it doesn't exist.
the `title` can't be undefined, so if it's not defined I assign to '' - typescript.

Reply:
Weird!